vignettes/GenomicFiles.Rmd

```diff
@@ -0,0 +1,738 @@
+using the parallel infrastructure provided in `BiocParallel`. Files and ranges
+are sent to workers along with MAP and REDUCE functions. The MAP extracts and/or
+manipulates data and REDUCE consolidates the results from MAP. Both MAP and
+REDUCE are executed in the distributed step and therefore reduction occurs on
+data from the same worker, not across workers.
+
+The chart in Figure 
+<a id="fig1" ></a>
+[1](#fig1)
+represents the division of labor in `reduceByRange`
+and `reduceRanges` with 3 files and 4 ranges. These functions split the problem
+by range which allows subsets (i.e., the same range) to be combined across
+different files. `reduceByRange` iterates through the files, invoking MAP and
+REDUCE for each range / file combination. This approach allows ranges extracted
+from the files to be kept separate or combined before the next call to `MAP`
+based on whether or not a `REDUCE` is supplied.
+
+`reduceRanges` applies `MAP` to each range / file combination and REDUCEs the
+output of all MAP calls. `REDUCE` usually plays a minor role by concatenating or
+unlisting results.
+
+```{r fig1, echo = FALSE, fig.cap = "Mechanics of `reduceByRange` and `reduceRanges`"}
+knitr::include_graphics("reduceByRange_flow.png")
+```
+
+In contrast to the 'byRange' approach, `reduceByFile` and `reduceFiles` (Figure
+<a id="fig2" ></a>
+[2](#fig2)) split the problem by file. Files are sent to
+different workers with the set of ranges allowing subsets (i.e., multiple
+ranges) from the same file to be combined. `reduceByFile` invokes MAP for each
+file / range combination allowing potential `REDUCE` after each MAP step.
+
+`reduceFiles` applies `MAP` to each range / file combination and REDUCEs the
+output of all MAP calls. `REDUCE` usually plays a minor role by concatenating or
+unlisting results.
+
+```{r fig2, echo = FALSE, fig.cap = "Mechanics of `reduceByFile` and `reduceFiles`"}
+knitr::include_graphics("reduceByFile_flow.png")
+```
+
+# Queries across files: `reduceByRange` and `reduceRanges`
+
+The `reduceByRange` and `reduceRanges` functions are designed for analyses that
+compare or combine data subsets across files. The first example in this section
+computes pileups on subsets from individual files then sums over all files. The
+second example computes coverage on a group of ranges for each file then
+performs a basepair-level $t$-test across files. The $t$-test example also
+demonstrates how to use a blocking factor to differentiate files by experimental
+group (e.g., case vs control).
+
+## Pileup summaries
+
+In this example nucleotide counts (pileups) are computed for the same
+ranges in each file (MAP step). Pileups are then summed by position
+resulting in a single table for each range across all files (REDUCE
+step).
+
+Create a GRanges with regions of interest:
+
+```{r pileups-ranges}
+gr <- GRanges("chr14", IRanges(c(19411677, 19659063, 105421963,
+     105613740), width=20))
```

Comment:
Usually, with R if the arguments span more than one line, we try to line it up with the start of the arguments. This is what they follow in the PDF for this section.